### PR TITLE
feat(flagd-core): update fractional operator to CBOR encoding (v3)

### LINKED ISSUE #324: [flagd] [FEATURE] Harden Hashing Consistency And Add Support For Non-string Attributes in Fractional Evaluation
Implement [this ADR](https://github.com/open-feature/flagd/blob/main/docs/architecture-decisions/fractional-non-string-rand-units.md) for the flagd provider.

### PR DESCRIPTION
**Pull request by m-olko**
## This PR
Update fractional bucketing in openfeature-flagd-core to use canonical CBOR encoding and unsigned 32-bit MurmurHash3 per the latest flagd specification.

- Implement number normalization for float/int consistency per specification

- Support non-string targeting keys and null bucketing keys returning None

- Use [flagKey, targetingKey] CBOR array for shorthand bucketing syntax

- Update flagd-testbed submodule to v3.10.1 and select @fractional-v3 tests

- Add unit tests for null handling, zero equivalence, float equivalence, and ordering

### Related Issues

Fixes #324

### CODE REVIEW

File: providers/openfeature-provider-flagd/tests/e2e/step/context_steps.py [hunk 1]
```diff
@@ -31,6 +31,8 @@ def update_context(
     evaluation_context: EvaluationContext, key: str, type_info: str, value: str
 ):
     """a context containing a key and value."""
+    if type_info == "String":
+        value = value.replace("\\\\", "\\")
```
[the remark this answers is not in the document]
Comment (m-olko) [reply providers/openfeature-provider-flagd/tests/e2e/step/context_steps.py:35]: This is a workaround for a known issue in pytest-bdd: https://github.com/pytest-dev/pytest-bdd/issues/769
It's already fixed on the main: https://github.com/pytest-dev/pytest-bdd/pull/820 but not yet released. Should I add a todo, and issue here on github to remove it when the library get's updated?

**Review by toddbaert — APPROVED** [review]
Looks good! Same as Java as far as I can tell, and testbed/tags match.

Just like Java, the only thing I would recommend is we agree on a release strategy. I started a slack conversation with you and @NeaguGeorgiana23
